dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/deduplication/DeduplicationController.java

```diff
@@ -181,6 +175,11 @@ public void mergePotentialDuplicate(
     {
         PotentialDuplicate potentialDuplicate = getPotentialDuplicateBy( id );
 
+        if ( potentialDuplicate == null )
+        {
+            throw new NotFoundException( "PotentialDuplicate with uid '" + id + "' not found." );
+        }
```

Comment:
getPotentialDuplicateBy is already catching NotFoundException if you look at the private  method

Reply:
Thanks for clearing that up; I'll revert it :)